### DISCUSSION TITLE
websocket issue

### DISCUSSION THREAD

**Discussion by xOluwaseyi** [vercel/next.js#41498] [Help]
I keep getting this error and I don't know what to do:

WebSocket connection to 'ws:<URL>/_next/webpack-hmr' failed: WebSocket is closed before the connection is established.

**Comment by icyJoseph** [comment]
Weird, got a repo to look at? or a Stackblitz reproduction?

I googled a bit, and surprisingly, restarting Windows (WLS) machines fixes the issue. Otherwise it seems that `assetPrefix`, might be causing issues.

**Reply by xOluwaseyi** [reply]
Maybe I'll put up a repo and share, I'll let you know when I do. Thank you

**Reply by tamalCodes** [reply]
> Weird, got a repo to look at? or a Stackblitz reproduction?
> 
> I googled a bit, and surprisingly, restarting Windows (WLS) machines fixes the issue. Otherwise it seems that `assetPrefix`, might be causing issues.

I am having the same problem, for local code,  sometimes it appears, sometimes it doesnot.
Plus auto-refresh doesnot works when the error is caused. I need to manually reload the page. 

It looks like magic at this point.

**Reply by tamalCodes** [reply]
Repo : https://github.com/IAmTamal/ShopCoders

**Reply by xOluwaseyi** [reply]
> > Weird, got a repo to look at? or a Stackblitz reproduction?
> > 
> > I googled a bit, and surprisingly, restarting Windows (WLS) machines fixes the issue. Otherwise it seems that `assetPrefix`, might be causing issues.
> 
> I am having the same problem, for local code,  sometimes it appears, sometimes it doesnot.
> Plus auto-refresh doesnot works when the error is caused. I need to manually reload the page. 
> 
> It looks like magic at this point.

I had no choice but to just ignore it since it's way beyond my control

**Reply by tamalCodes** [reply]
Well ignoring won't work either. It's a pain to refresh again and again manually.  IDK what's causing it, nor am i able to find a fix.

**Reply by xOluwaseyi** [reply]
Oh wait, is it actually the reason why I had to refresh manually a lot sometimes? 🧐 @IAmTamal 

The fast refresh only worked whenever it felt like working

**Reply by tamalCodes** [reply]
Yes, once there's no error for websocket the fast-refresh works well !

**Reply by xOluwaseyi** [reply]
I see. Okay. I wonder when the issue will get fixed because it always seems like there's a bug from the dev when they encounter it the first time which makes them start looking for something that's not lost and wasting valuable time.

**Reply by tamalCodes** [reply]
Exactly my point !
